language/testing-infra/diem-transactional-test-runner/src/diem_test_harness.rs

```diff
@@ -0,0 +1,320 @@
+        ScriptFunction as TransactionScriptFunction,
+    },
+};
+use language_e2e_tests::executor::FakeExecutor;
+use move_binary_format::file_format::{CompiledModule, CompiledScript};
+use move_core_types::{
+    account_address::AccountAddress,
+    gas_schedule::{GasAlgebra, GasConstants},
+    identifier::IdentStr,
+    language_storage::{ModuleId, TypeTag},
+    transaction_argument::{convert_txn_args, TransactionArgument},
+};
+use move_lang::{shared::verify_and_create_named_address_mapping, FullyCompiledProgram};
+use move_transactional_test_runner::{
+    framework::{run_test_impl, CompiledState, MoveTestAdapter},
+    tasks::{EmptyCommand, InitCommand, SyntaxChoice, TaskInput},
+};
+use once_cell::sync::Lazy;
+use std::collections::BTreeMap;
+use std::path::Path;
+use structopt::StructOpt;
+
+enum DiemTestCommand {
+    CreateAccount,
+}
+
+struct DiemTestAdapter<'a> {
+    compiled_state: CompiledState<'a>,
+    exec: FakeExecutor,
+    default_syntax: SyntaxChoice,
+}
+
+/// A set of common parameters required to create transactions.
+struct TransactionParameters {
+    pub sequence_number: u64,
+    pub max_gas_amount: u64,
+    pub gas_unit_price: u64,
+    pub gas_currency_code: String,
+    pub expiration_timestamp_secs: u64,
+}
+
+fn get_default_transaction_parameters(
+    exec: &FakeExecutor,
+    signer_addr: &AccountAddress,
+) -> TransactionParameters {
+    // TODO: get rid of the unwrap and handle the case where the account does not exist
+    let account_resource = exec.read_account_resource_at_address(signer_addr).unwrap();
+    let sequence_number = account_resource.sequence_number();
+    let gas_currency_code = XUS_NAME.to_owned();
+    let max_number_of_gas_units = GasConstants::default().maximum_number_of_gas_units;
+    let gas_unit_price = 0;
+    let max_gas_amount = if gas_unit_price == 0 {
+        max_number_of_gas_units.get()
+    } else {
+        let account_balance = exec
+            .read_balance_resource_at_address(
+                signer_addr,
+                account_config::from_currency_code_string(&gas_currency_code).unwrap(),
+            )
+            .unwrap_or_else(|| {
+                panic!(
+                    "Couldn't read balance of type {:?} for account {:?}",
+                    signer_addr, gas_currency_code
+                )
+            });
+        std::cmp::min(
+            max_number_of_gas_units.get(),
+            account_balance.coin() / gas_unit_price,
+        )
+    };
+
+    TransactionParameters {
+        sequence_number,
+        gas_currency_code,
+        gas_unit_price,
+        max_gas_amount,
+        expiration_timestamp_secs: 40000,
+    }
+}
+
+fn parse_ed25519_private_key(s: &str) -> Result<Ed25519PrivateKey> {
+    Ok(Ed25519PrivateKey::from_encoded_string(s)?)
+}
+
+#[derive(StructOpt, Debug)]
+struct DiemPublishArgs {
+    #[structopt(short = "k", long = "private-key", parse(try_from_str = parse_ed25519_private_key))]
```

Comment:
Do we need to support this? I'd rather see it automanaged. We are testing the DF, not key management :)

Reply:
Well a private key is something you MUST provide in order to execute a signed transaction, just like the real blockchain. We can make the test adapter maintain a key mapping, similar to what we have in the old functional tests, but I thought we decided to favor explicitness and not do this based on our previous discussions?